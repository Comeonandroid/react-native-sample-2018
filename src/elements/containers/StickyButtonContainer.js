import React from 'react'
import styled from 'styled-components'
import { PropTypes } from 'prop-types'

import { View } from '../index'

const ButtonContainer = ({ children, ...rest }) => (
  <Element {...rest}>
    {Array.isArray(children) ? children.map( (el, i) =>
      <Cell key={i}>{el}</Cell>
    ) : children}
  </Element>
)

ButtonContainer.propTypes = {
  children: PropTypes.node
}

const Element = styled(View) `
  width: 100%;
  height: 40px;
  position: absolute;
  bottom: 0;
  flex-direction: row;
`

const Cell = styled(View) `
  flex: 1;
  borderRightWidth: 1px;
  borderRightColor: #554225;

  &:last-of-type {
    border: 0;
  }
`

export default ButtonContainer
