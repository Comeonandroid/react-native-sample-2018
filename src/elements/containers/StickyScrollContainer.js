import React from 'react'
import styled from 'styled-components'
import { PropTypes } from 'prop-types'

import { ScrollContainer } from '../index'

const StickyScrollContainer = ({ children, ...rest }) => (
  <Element {...rest}>
    {children}
  </Element>
)

StickyScrollContainer.propTypes = {
  children: PropTypes.node
}

const Element = styled(ScrollContainer) `
  padding-bottom: 40px;
`

export default StickyScrollContainer
