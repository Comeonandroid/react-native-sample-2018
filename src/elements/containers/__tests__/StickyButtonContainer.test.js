import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'
import { itBehavesLike } from '../../../specs/shared_examples/index'

import StickyScrollContainer from '../StickyScrollContainer'
import StickyButtonContainer from '../StickyButtonContainer'

describe('<StickyButtonContainer />', () => {
  itBehavesLike('aThemedComponent', { subject: <StickyButtonContainer /> })
  itBehavesLike('aComponentWithChildren', { subject: <StickyButtonContainer /> })

  it('have height equals container padding', () => {
    const containerTree = renderer.create(<StickyScrollContainer />).toJSON()
    const buttonTree = renderer.create(<StickyButtonContainer />).toJSON()
    const containerPadding = containerTree.props.style[1][1][0].paddingBottom
    const buttonHeight = buttonTree.props.style[1][0].height

    expect(buttonHeight).toEqual(containerPadding)
  })
})
