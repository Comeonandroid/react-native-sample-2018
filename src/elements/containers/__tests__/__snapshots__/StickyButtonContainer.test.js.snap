// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StickyButtonContainer /> dark: renders correctly 1`] = `
<View
  style={
    Array [
      Object {},
      Array [
        Object {
          "bottom": 0,
          "flexDirection": "row",
          "height": 40,
          "position": "absolute",
          "width": "100%",
        },
        undefined,
      ],
    ]
  }
/>
`;

exports[`<StickyButtonContainer /> light: renders correctly 1`] = `
<View
  style={
    Array [
      Object {},
      Array [
        Object {
          "bottom": 0,
          "flexDirection": "row",
          "height": 40,
          "position": "absolute",
          "width": "100%",
        },
        undefined,
      ],
    ]
  }
/>
`;

exports[`<StickyButtonContainer /> renders component when passed in 1`] = `
<Styled(View)>
  <Component />
</Styled(View)>
`;

exports[`<StickyButtonContainer /> renders text when passed in 1`] = `
<Styled(View)>
  Test
</Styled(View)>
`;
